ci: update Postgres version

eulerto opened the pull request:
Use the latest stable version of PostgreSQL. It covers the latest protocol changes.

eulerto committed:
ci: update Postgres version

Use the latest stable version of PostgreSQL. It covers the latest
protocol changes.

eulerto committed:
ci: update Postgres version

Use the latest stable version of PostgreSQL. It covers the latest
protocol changes.

JelteF commented:
The OSX build seems broken because of this.

eulerto commented:
It seems the setup errors are gone. Hmm. I don't know why this test is failing like this. @JelteF do you have an idea?

JelteF commented:
The first error I've seen before when not using the `master` version of `psycopg`. Now that psycopg 3.2 is out though there's no need to use the master version of it anymore. So I think it makes sense to change it to install the released version in `requirements.txt` and then change all the tests that break because we are doing so.

eulerto committed:
psycopg 3.2 is out

psycopg from master is not required anymore. The referred version is
compatible with PgBouncer.

https://www.psycopg.org/psycopg3/docs/news.html#psycopg-3-2

eulerto committed:
psycopg 3.2 is out

psycopg from master is not required anymore. The referred version is
compatible with PgBouncer.

https://www.psycopg.org/psycopg3/docs/news.html#psycopg-3-2

eulerto committed:
Blindly try to fix psycopg setup

It seems libpq cannot be found by psycopg.

eulerto committed:
Blindly try to fix psycopg setup

It seems libpq cannot be found by psycopg.

eulerto committed:
CI: update macOS image

According to Cirrus documentation [1], there is only one image available
for macOs. Let's use it.

[1] https://cirrus-ci.org/guide/macOS/

eulerto committed:
CI: update macOS image

According to Cirrus documentation [1], there is only one image available
for macOs. Let's use it.

[1] https://cirrus-ci.org/guide/macOS/

JelteF committed:
Try some more env variables for MacOS

JelteF committed:
Try some more env variables for MacOS

eulerto committed:
Blindly try to fix, take 2

Fix the previous attempt.

eulerto committed:
Blindly try to fix, take 2

Fix the previous attempt.

eulerto committed:
Add pg_config to the PATH

It is an attempt to find libpq required by psycopg. It also adds a debug
aid to list all of the postgresql files (if the path is not correct).

eulerto committed:
Add pg_config to the PATH

It is an attempt to find libpq required by psycopg. It also adds a debug
aid to list all of the postgresql files (if the path is not correct).

eulerto committed:
fix an extra slash

eulerto committed:
fix an extra slash

eulerto committed:
Undo the previous command

Instead export the PATH in a separate command.

eulerto committed:
Undo the previous command

Instead export the PATH in a separate command.

eulerto committed:
Move PATH to env section

It also removes env variables that are not required (?).

eulerto committed:
Move PATH to env section

It also removes env variables that are not required (?).

eulerto committed:
OpenSSL is required

eulerto committed:
OpenSSL is required

eulerto committed:
Merge branch 'master' into ci-pg16

eulerto committed:
Merge branch 'ci-pg16' of github.com:eulerto/pgbouncer into ci-pg16

eulerto committed:
Remove PKG_CONFIG_PATH

eulerto committed:
Remove LD_LIBRARY_PATH

eulerto committed:
Merge branch 'master' into ci-pg16

eulerto commented:
I'm fine with the current PR. I fixed the Mac OS. The alpine is still broken but it is not because of this PR but another issue with TLS. @JelteF objections?

JelteF commented on .cirrus.yml:
```diff
@@ -195,25 +195,26 @@ task:
     - ./autogen.sh
     - ./configure --prefix=$HOME/install --enable-werror
     - make -j4
+  env:
+    PATH: /opt/homebrew/opt/postgresql@${PGVERSION}/bin:$PATH
```
can this be merged with the other `env` section?

eulerto replied:
I think so. The main reason I added here was that that path exists only after installing the PostgreSQL package.

JelteF replied:
Let's merge them then, because it's a bit confusing that we have two `env` keys in the same object. Based on some quick searching it doesn't seem specified what that means according the yaml spec (it seems like github merges the objects)

JelteF commented on .cirrus.yml:
```diff
@@ -195,25 +195,26 @@ task:
+  env:
+    PATH: /opt/homebrew/opt/postgresql@${PGVERSION}/bin:$PATH
   test_script:
-    - source venv/bin/activate && make check -j4 CONCURRENCY=4
+    - source venv/bin/activate && make -j4 check CONCURRENCY=4
```
Was this change actually needed?

eulerto replied:
There is no strong reason. Consistency with the similar commands. I should have mentioned it.

JelteF reviewed (approved):
LGTM appart from two small comments. Please note that we should update the required jobs in the github repo settings before (or quickly after) merging.

eulerto commented:
> Please note that we should update the required jobs in the github repo settings before (or quickly after) merging.

What settings are you referring to? AFAICS we are not using GH Actions. Cirrus CI is a third-part app that is integrated into GitHub.

JelteF commented:
> What settings are you referring to?

I the "Required" tag after the CI job: 
![image](https://github.com/user-attachments/assets/f96ea040-be01-431a-8111-a69e0dd6928a)

This checks are required for merging into the master branch, but they contain the 14 number. They can be changed in the branch protection settings here: https://github.com/pgbouncer/pgbouncer/settings/branch_protection_rules/27973333

JelteF committed:
Merge env

JelteF commented:
FYI: I made the change I suggested to the `env` entry, merged the PR and updated the branch protection rules
